aikau/src/main/resources/alfresco/lists/SelectedItemStateMixin.js

```diff
@@ -0,0 +1,261 @@
+/**
+ * Copyright (C) 2005-2015 Alfresco Software Limited.
+ *
+ * This file is part of Alfresco
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * <p>This module can be mixed into widgets that need to maintain the state of any selected items. Item
+ * selection is typically actioned by either individual [Selector]{@link module:alfresco/renderers/Selector}
+ * widgets in a [list]{@link module:alfresco/lists/AlfList} or by a menu action such as the ones provided by
+ * the [AlfSelectDocumentListItems]{@link module:alfresco/documentlibrary/AlfSelectDocumentListItems}.</P>
+ * <p>Mixing modules should call the 
+ * [createSelectedItemSubscriptions]{@link module:alfresco/lists/SelectedItemStateMixin#createSelectedItemSubscriptions}
+ * on creation to keep track of selected items. The can also call the 
+ * [publishSelectedItems]{@link module:alfresco/lists/SelectedItemStateMixin#publishSelectedItems} function
+ * to broadcast the details of changes to item selection.</p>
+ * 
+ * @module alfresco/lists/SelectedItemStateMixin
+ * @author Dave Draper
+ * @since 1.0.39
+ */
+define(["dojo/_base/declare",
+        "alfresco/core/Core",
+        "alfresco/core/topics",
+        "alfresco/documentlibrary/_AlfDocumentListTopicMixin",
+        "dojo/_base/lang"], 
+        function( declare, Core, topics, _AlfDocumentListTopicMixin, lang) {
+   
+   return declare([Core, _AlfDocumentListTopicMixin], {
+      
+      /**
+       * This is used to keep track of the documents that are currently selected. It is initialised to an empty
+       * array in the constructor, [onItemSelected]{@link module:alfresco/lists/SelectedItemStateMixin#onItemSelected} 
+       * adds elements and the [onItemDeselected]{@link module:alfresco/lists/SelectedItemStateMixin#onItemDeselected} 
+       * removes them.
+       *
+       * @instance
+       * @type {object}
```

Comment:
Then you need to correct the JSDoc comment just above that says it's initialised to an empty array! :)